File: 프로그래머스/lv0/120894. 영어가 싫어요/영어가 싫어요.cs
using System;

public class Solution {
    
    private string[] numStrList = new string[10]{"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    
    public long solution(string numbers) {
        
        string convertedStr = numbers;
        for(int i = 0; i < numStrList.Length; i++)
            convertedStr = convertedStr.Replace(numStrList[i], i.ToString());
        
        return long.Parse(convertedStr);
    }
}

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

long long solution(string numbers) {

    map<string, string> _map
    { 
        {"zero", "0"},
        {"one", "1"},
        {"two", "2"},
        {"three", "3"},
        {"four", "4"},
        {"five", "5"},
        {"six", "6"},
        {"seven", "7"},
        {"eight", "8"},
        {"nine", "9"},
    };
    
    for (const auto& pair : _map)
    {
        size_t pos;
        do
        {
            pos = numbers.find(pair.first);
            if (pos != string::npos)
                numbers.replace(pos, pair.first.size(), pair.second);
        }
        while (pos != string::npos);
    }
    
    return stoll(numbers);
}